bigquery/google/cloud/bigquery/client.py

```diff
@@ -567,12 +632,54 @@ def update_model(self, model, fields, retry=DEFAULT_RETRY):
         )
         return Model.from_api_repr(api_response)
 
+    def update_routine(self, routine, fields, retry=DEFAULT_RETRY):
```

Comment:
Should we have something in the docstring about partial update support b/134928435?